internal/mq/mqimpl/rocksmq/client/producer_impl.go

```diff
@@ -50,11 +51,23 @@ func (p *producer) Topic() string {
 }
 
 // Send produce message in rocksmq
-func (p *producer) Send(message *ProducerMessage) (UniqueID, error) {
+func (p *producer) Send(message *mqwrapper.ProducerMessage) (UniqueID, error) {
+	// NOTICE: this is the hack.
+	// we should not unmarshal the payload here but we can not extend the payload byte
+	payload := message.Payload
+	header, err := UnmarshalPayload(message.Payload)
+	if err == nil && header != nil && header.Base != nil {
+		// try to marshal properties into message if message is real message
+		header.Base.Properties = message.Properties
+		payload, err = MarshalPayload(header)
```

Comment:
Using a marshaled header as a message payload? i

Reply:
This is a hack for compatibility.
All the payload should implement messagebase.

If not and marshal is error, then we fallback and don't serialize properties.

Properties contains tracing datas for now